docs/core/compatibility/sdk/7.0/solution-level-output-no-longer-valid.md

```diff
@@ -37,7 +39,33 @@ This breaking change may require modifications to build scripts and continuous i
+
+### Single project, multiple TargetFrameworks
+
+Now imagine a solution that contains a single project with multiple `TargetFrameworks`, `net6.0` and `net7.0`. Because of multi-targeting, the project will be build twice, once for each `TargetFramework`. For each of these 'inner' builds the `OutputPath` will be set to the same value, and so the outputs for each of the inner builds will be placed in the same directory. This means that whichever build completes last will overwrite the outputs of the other build, and in a parallel-build system like MSBuild operates in by default, 'last' is indeterminate.
+
+### Library => Console => Test, single TargetFramework
+
+Now imagine a solution that contains a library project, a console project that references the library project, and a test project that references the console project. All of these projects target a single `TargetFramework`, `net7.0`. In this case, the library project will be built first, and then the console project will be built. The test project will be built last, and will reference the console project. For each built project, the outputs of each build will be copied into the directory specified by the `OutputPath`, and so the final directory will contain assets from all three projects. This works for testing, but for publishing may result in test assets being sent to production.
+
+### Library => Console => Test, multiple TargetFrameworks
+
+Now take the same chain of projects and add a `net6.0` `TargetFramework` build to them in addition to the `net7.0` build. The same problem as the single-project, multi-targeted build occurs - inconsistent copying of TFM-specific assets to the specified directory.
+
+### Multiple apps
+
+So far we have been looking at scenarios with a linear dependency graph - but many solutions may contain multiple related applications. This raises a new way to get inconsistent outputs in the `OutputPath` - inconsistent resolution of `PackageReference`s across projects.  NuGet helps ensure that within a project and its' `ProjectReference`s, any `PackageReference`s and transitive dependencies (meaning dependencies of `PackageReference`s declared in a project file) are unified to the same version. Because the unification is done within the context of a single project and its dependent projects, this means it is possible to resolve different versions of a package when two separate top-level projects are built. If the versions resolved in this case are different enough, then runtime errors can occur.
+
+This can also happen if you have different explicit versions of `PackageReference`s in the top-level projects of the solution. Because a solution-level build or publish will copy all assets into a single directory, whatever project was copied last will decide what version of the `PackageReference` is actually in use, especially if the entire `OutputPath` is then deployed to a production environment.
```

Comment:
```suggestion
So far we have been looking at scenarios with a linear dependency graph - but many solutions may contain multiple related applications.  This means that multiple apps may be built concurrently to the same output folder.  If the apps include a dependency file with the same name, then the build may intermittently fail when multiple projects try to write to that file in the output path concurrently.

If multiple apps depend on different versions of a file, then even if the build succeeds, which version of the file is copied to the output path may be non-deterministic.  This can happen when the projects depend (possibly transitively) on different versions of a NuGet package.  Within a single project, NuGet helps ensure that its dependencies (including any transitive dependencies through NuGet packages and/or project references) are unified to the same version.  Because the unification is done within the context of a single project and its dependent projects, this means it is possible to resolve different versions of a package when two separate top-level projects are built.  If the project that depends on the higher version copies the dependency last, then often the apps will run successfully.  However, if the lower version is copied last, then the app that was compiled against the higher version will fail to load the assembly at runtime.  Because the version that is copied can be non-deterministic, this can lead to sporadic, unreliable builds where it is very difficult to diagnose the issue.
```